### PR TITLE
Fontsize

### PR DESCRIPTION
**Pull request by Moeez905**
Adjusted the font size of hamburger menu as mentioned by @thc202 

Part of #3.

### CODE REVIEW

**Comment by thc202** [comment]
For correctness, it was not me who mentioned that, the original issue was raised by psiinon.

**Comment by psiinon** [comment]
Before:
Screenshot 2023-05-09 at 14-57-06 The ZAP Homepage

After:
Screenshot 2023-05-09 at 14-58-02 The ZAP Homepage

**Comment by kingthorin** [comment]
Do these styles only apply to mobile?
Are the rest of our styles font-sizes defined in `em`?

(Note: I'm not a CSS or front-end person, so I totally could have overlooked simple answers.)

**Comment by Moeez905** [comment]
> Do these styles only apply to mobile? Are the rest of our styles font-sizes defined in `em`?
> 
> (Note: I'm not a CSS or front-end person, so I totally could have overlooked simple answers.)

Yes they apply to mobile and font-size is in "px"
I just found an error in my code because of which ZAP logo itself also got smaller too. I'll look into it.

**Comment by kingthorin** [comment]
I know what you added is in `px`, but my question was more on the topic of consistency :wink: What do existing font-size declarations use? Or is it already mixed `em` and `px`.

**Comment by Moeez905** [comment]
Yes, you're right! format of my font-size declarations isn't consistent with other relevant declarations, they're in "em" while I did it in "px"
I'll fix my code.
Thanks,

File: src/css/_navigation.scss [hunk 1]
```diff
@@ -125,6 +127,8 @@
   background-clip: padding-box; 
   font-size: 0.9em;
   transition: all 40ms cubic-bezier(0.42, 0, 1, 1);
+
```
Comment (psiinon) [inline src/css/_navigation.scss:130]: The blank lines can be reverted 😁

File: src/css/components/_button.scss [hunk 2]
```diff
@@ -13,6 +13,7 @@
 
 .button--blue {
   background: var(--blue);
+  margin-bottom: 10px;
```
Comment (psiinon) [inline src/css/components/_button.scss:16]: Has this been pulled in from the previous PR?

File: src/css/_navigation.scss (hunk 1 above)
Comment (Moeez905) [reply src/css/_navigation.scss]: Reverted!

File: src/css/components/_button.scss (hunk 2 above)
Comment (Moeez905) [reply src/css/components/_button.scss]: Oh yes, I should've created a new branch from the master branch instead of "gap branch".

**Comment by njmulsqb** [comment]
> Yes, you're right! format of my font-size declarations isn't consistent with other relevant declarations, they're in "em" while I did it in "px" I'll fix my code. Thanks,

The font size is still in `px`, you need to replace `px` with `em`

**Comment by Moeez905** [comment]
> > Yes, you're right! format of my font-size declarations isn't consistent with other relevant declarations, they're in "em" while I did it in "px" I'll fix my code. Thanks,
> 
> The font size is still in `px`, you need to replace `px` with `em`

Thanks!
Font size changed to "em" and now mobile-view has a different/slightly smaller font size as compared to desktop view because desktop-view font size seems very small if only 1 size is used sitewide.
before
after

**Comment by Moeez905** [comment]
Hey, it's been a while since I've pushed the changes, can we merge the PR and move towards working on more issues?

File: src/css/_navigation.scss [hunk 3]
```diff
@@ -220,6 +230,7 @@ input {
   #search-menu {
     width: 100%;
     line-height: 0;
+    font-size: 25px;
```
Comment (kingthorin) [inline src/css/_navigation.scss:233]: Why can’t this one be em like others?

File: src/css/_navigation.scss [hunk 4]
```diff
@@ -73,15 +73,25 @@
   position: relative;
   top:-4px;
 }
-
-
```
Comment (kingthorin) [inline src/css/_navigation.scss:77]: As @psiinon mentioned earlier the white space changes should be reverted

File: src/css/_navigation.scss (hunk 3 above)
Comment (Moeez905) [reply src/css/_navigation.scss]: updated

File: src/css/_navigation.scss (hunk 4 above)
Comment (njmulsqb) [reply src/css/_navigation.scss:77]: @Moeez905 here you actually need to bring back the white spaces that have been removed by the extension you used, `-` sign in red denotes an empty space that has been removed by one of your commits

**Comment by Moeez905** [comment]
> Please rebase instead of merging.
> 
> ```shell
> git fetch upstream
> git rebase upstream/master
> # Fix the white space changes, turn off whatever in your IDE keeps applying them
> git commit -a --amend --no-edit --signoff
> git push --force-with-lease
> ```

Thank You! 
Updated

File: src/css/_navigation.scss [hunk 5]
```diff
@@ -80,8 +80,20 @@
...
+    font-size: 0.7em;
+  }
+}
+@media (min-width: 576px) and (max-width: 767.98px) {
+  /* CSS styles for small devices */
+  .site-header nav a {
+      font-size: 0.7em;
+    }
 }
-
```
Comment (kingthorin) [inline src/css/_navigation.scss:84]: Restore

File: src/css/_navigation.scss [hunk 6]
```diff
@@ -97,7 +109,7 @@
   height: 24px;
 }
 
-.hamburger-icon {}
+
```
Comment (kingthorin) [inline src/css/_navigation.scss:112]: Remove

File: src/css/_navigation.scss [hunk 7]
```diff
@@ -134,10 +146,7 @@
   }
 }
 
-.site-header nav .sub-menu {
-  a {
-  }
-}
+
```
Comment (kingthorin) [inline src/css/_navigation.scss:149]: Remove

File: src/css/_navigation.scss [hunk 8]
```diff
@@ -236,7 +246,7 @@ input {
       width: 100%;
     }
   }
-  
+
```
Comment (kingthorin) [inline src/css/_navigation.scss:249]: Remove (a single blank line is all that’s necessary here)

File: src/css/components/_button.scss [hunk 9]
```diff
@@ -13,7 +13,6 @@
 
 .button--blue {
   background: var(--blue);
-  margin-bottom: 10px;
```
Comment (kingthorin) [inline src/css/components/_button.scss:16]: Wasn’t this added by something that was already merged?

Comment (Moeez905) [reply src/css/components/_button.scss:16]: Sorry, I didn't get you! But this line of code isn't present in my latest pull(sync fork) too, meanwhile, I added it again and removed mentioned unnecessary spaces, you may have a look

Comment (kingthorin) [reply src/css/components/_button.scss:16]: It was added in this pull of yours: https://github.com/zaproxy/zaproxy-website/pull/1758/files why would we remove it in this one?

Comment (Moeez905) [reply src/css/components/_button.scss:16]: it wasn't supposed to be removed I think it was a mistake, So I just restored it & a couple of other fixes too.
sorry for petty mistakes 😅, 
gaps are now pretty much adjusted. Still, if there's any left, let me know!

Comment (kingthorin) [reply src/css/components/_button.scss:16]: You haven't pushed your changes.

Comment (Moeez905) [reply src/css/components/_button.scss]: > You haven't pushed your changes.

done

**Comment by njmulsqb** [comment]
Hi @Moeez905 
Your commits are not signed properly, click on the `details` button and follow the command it gives you.
image
Plus, some of the changes requested by Rick are not implemented, look for the review changes above that does not include the **Outdated** tag

**Comment by Moeez905** [comment]
> Hi @Moeez905 Your commits are not signed properly, click on the `details` button and follow the command it gives you. image Plus, some of the changes requested by Rick are not implemented, look for the review changes above that does not include the **Outdated** tag

thanks! its done now

**Comment by Moeez905** [comment]
I have removed/restored more white spaces, some highlighted ones are those that are separating CSS code but still they can be removed so let me know

**Comment by kingthorin** [comment]
The PR should include all lay those changes that pertain to the issue at hand.
If formatting changes are needed/wanted they should be done in a separate PR, along with the configs/workflows for their enforcement.

File: src/css/_navigation.scss [hunk 10]
```diff
@@ -209,7 +225,7 @@ input {
       width: 100%;
       padding: 0;
       margin:0;
-
```
Comment (Moeez905) [inline src/css/_navigation.scss:212]: I added a space here but it was showing as an additional space, removing it now shows red sign. What is it expecting? @kingthorin

**Review by Moeez905 — COMMENTED** [review]
I think I have removed the spaces *sigh* please merge this PR or let me know if there's still something left? :')

File: src/css/_navigation.scss (hunk 10 above)
Comment (thc202) [reply src/css/_navigation.scss:212]: It's missing spaces.

**Comment by kingthorin** [comment]
There is. I’ll take care of it.

**Review by kingthorin — APPROVED** [review]

**Comment by kingthorin** [comment]
⚠️ Please do not make any further changes to this PR ⚠️ 

If something else comes up I'll take care of it.

**Comment by Moeez905** [comment]
Sure! Thanks

**Comment by thc202** [comment]
@kingthorin can you rebase?

**Comment by kingthorin** [comment]
Sure.

**Comment by kingthorin** [comment]
Done

**Comment by thc202** [comment]
The "Search..." now seems to be way to big.
Before:
before

After:
after

Was that one purpose?

**Comment by kingthorin** [comment]
Thanks, I'll get that tweaked.

**Comment by kingthorin** [comment]
Tweaked, before and after:
image image

File: src/css/_navigation.scss [hunk 11]
```diff
@@ -80,6 +80,22 @@
   font-weight: bold;
   text-decoration: none;
   transition: all 100ms ease;
+  font-size: 1em;
+}
+
+@media (max-width: 575.98px) {
+
+  /* CSS styles for extra small devices */
```
Comment (thc202) [inline src/css/_navigation.scss:88]: This and the following should be removed.

File: src/css/_navigation.scss [hunk 12]
```diff
@@ -80,6 +80,22 @@
...
+  .site-header nav a {
+    font-size: 0.7em;
+  }
+}
+
+@media (min-width: 576px) and (max-width: 767.98px) {
```
Comment (thc202) [inline src/css/_navigation.scss:94]: This has the same styles as the above, why not have a single condition?

**Comment by kingthorin** [comment]
Agreed, adjusted.

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you both!

**Review by psiinon — APPROVED** [review]

**Pull request merged**